### ISSUE TITLE
[PRE REVIEW]: geopolygon: retrieve and correctly reshape the polygon of a geographical area

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1895]
**Submitting author:** @CostRagno (<a href="http://orcid.org/0000-0002-8807-4586">Costantino  Ragno</a>)
**Repository:** <a href="https://github.com/CostRagno/geopolygon" target ="_blank">https://github.com/CostRagno/geopolygon</a>
**Version:** v0.1
**Editor:** Pending
**Reviewer:** Pending

**Author instructions**

Thanks for submitting your paper to JOSS @CostRagno. **Currently, there isn't an JOSS editor assigned** to your paper.

@CostRagno if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```


**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1895 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-cd1f92b00da7/lib/whedon.rb:114:in `editor': undefined method `[]' for nil:NilClass (NoMethodError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-cd1f92b00da7/lib/whedon/compilers.rb:113:in `pdf_from_markdown'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-cd1f92b00da7/lib/whedon/compilers.rb:13:in `generate_pdf'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-cd1f92b00da7/bin/whedon:56:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-cd1f92b00da7/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by labarba** [comment]
@openjournals/dev -- S.O.S. All paper compilations are failing today!

**Comment by labarba** [comment]
@CostRagno — I'm going to send this back to you for some work on the paper, which should not be a copy-paste of the README, but a narrative that explains the software functionality to a non-specialist, places the software in the general state of the art (e.g., any other software out there doing similar things?), and illustrates the research context where it might be used. Please take another look at the JOSS [Author Guide](https://joss.readthedocs.io/en/latest/submitting.html).

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1895 with the following error: 

 Error reading bibliography ./paper.bib (line 24, column 3):
unexpected "a"
expecting space, ",", white space or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by arfon** [comment]
@CostRagno - this PR should fix your paper compilation: https://github.com/CostRagno/geopolygon/pull/1

**Comment by CostRagno** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1895 with the following error: 

 [WARNING] Could not convert image '/tmp/tex2pdf.-d49d3a75c2cdd527/8370c256978f712e7ec28c3673c28cd1f86eca78.shtml': Cannot load file
  Jpeg Invalid marker used
  PNG Invalid PNG file, signature broken
  Bitmap Invalid Bitmap magic identifier
  GIF Invalid Gif signature : 
  
  
  
  
  
  
  HDR Invalid radiance file signature
  Tiff Invalid endian tag value
  TGA Invalid bit depth (104)
[WARNING] Could not convert image '/tmp/tex2pdf.-d49d3a75c2cdd527/90140b843f9e82182444dd035d63f0d168ab2aff.shtml': Cannot load file
  Jpeg Invalid marker used
  PNG Invalid PNG file, signature broken
  Bitmap Invalid Bitmap magic identifier
  GIF Invalid Gif signature : 
  
  
  
  
  
  
  HDR Invalid radiance file signature
  Tiff Invalid endian tag value
  TGA Invalid bit depth (104)
Error producing PDF.
! LaTeX Error: Cannot determine size of graphic in /tmp/tex2pdf.-d49d3a75c2cdd5
27/8370c256978f712e7ec28c3673c28cd1f86eca78.shtml (no BoundingBox).

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.451 ...56978f712e7ec28c3673c28cd1f86eca78.shtml}

Looks like we failed to compile the PDF

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.1016/j.dss.2018.03.004 may be missing for title: Business social media analytics: Characterization and conceptual framework
- https://doi.org/10.1016/j.dss.2019.03.006 may be missing for title: Twitter user geolocation using web country noun searches
- https://doi.org/10.5220/0002080800610068 may be missing for title: Concave hull: A k-nearest neighbours approach for the computation of the region occupied by a set of points

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@CostRagno - this PR should fix your figures: https://github.com/CostRagno/geopolygon/pull/2

**Comment by CostRagno** [comment]
@labarba thank you for your comments. We will work more on the paper based on your suggestions :)

**Comment by CostRagno** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01895/joss.01895/10.21105.joss.01895.pdf)

**Comment by CostRagno** [comment]
@arfon now it is perfectly working. Thank you very much. We had problems because the https://whedon.theoj.org/ do not generate the pdf and because it was our first joss submission.

**Comment by danielskatz** [comment]
@CostRagno - is this paper ready to be moved on to review in your opinion?  In other words, have you finished what you intended when you wrote the following?

> We will work more on the paper based on your suggestions

**Comment by danielskatz** [comment]
Hi @CostRagno - I've consulted with the JOSS editorial team and we've concluded that this submission is not in scope for JOSS as it falls into our 'minor utility' category.

For increased visibility, we would encourage you to consider contributing the code for polygon ordering upstream to the [geopandas](https://github.com/geopandas/geopandas) library.

Alternatively, the general usability of the software could be improved by separating the algorithm for ordering the polygons (with focus on compatibility with geopandas and the rest of the Python geospatial stack) from the data fetching and specific use case of OpenStreetMap data. 

Thanks again for your interest in JOSS.
